keys/source.go

```diff
@@ -0,0 +1,53 @@
+package keys
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Source represents one key source
+type Source interface {
+	GetKey(ctx context.Context) ([]byte, error)
+}
+
+// Sources represents multiple key sources
+type Sources []Source
+
+// GetKey iterates sources and returns the first successful response
+func (sources Sources) GetKey(ctx context.Context) ([]byte, error) {
+	for _, source := range sources {
+		bts, err := source.GetKey(ctx)
+		if err == nil {
+			return bts, nil
+		}
+	}
+	return nil, ErrGetKeySource{"no sources could be resolved"}
+}
+
+// HTTPSource is a key with a URL source over HTTP
+type HTTPSource string
+
+// GetKey retrieves the key from the source over HTTP
+func (source HTTPSource) GetKey(ctx context.Context) ([]byte, error) {
+	req, err := http.NewRequest(http.MethodGet, string(source), nil)
+	if err != nil {
+		return nil, ErrGetKeySource{}
+	}
+	req = req.WithContext(ctx)
+	client := &http.Client{}
```

Comment:
You really ought to be able to pass in a client as a parameter here, other parts of the core allow for this (transport, for example).

Instead use a default client if nil is passed in

Reply:
The problem is that if you pass an HTTP client you would have to break the GetKey interface. What is the reason you would want to pass your own HTTP client? We do already allow passing a context to handle timeouts and whatnot.